client/src/index.jsx

```diff
@@ -0,0 +1,206 @@
+          ratings: mainProductData[0].ratings,
+          title: mainProductData[0].title
+        })
+      })
+    }
+  
+    getPhotos() {
+      axios.get('/api/photos').then((data) => {
+        let photourldata = data.data;
+        console.log(photourldata, 'line 52')
+        let photourlarray = [];
+        // makes a random number from 0 to 9
+        let test = Math.floor(Math.random() * 10);
+        photourldata = photourldata.filter(element => element.product_id === test)
+        console.log(photourldata, "logging data from api photos call");
+        // TODO: access string inside object
+        photourldata.forEach(element => {photourlarray.push(element.photo_url)
+        console.log(photourlarray, 'logging data from photourldata line 26')
+        });
+        // TODO: refactor filter method of getting correct urls
+          this.setState({
+            photos: photourlarray
+          });
+      })
+    }
+
+    handleModal(e) {
+      console.log('we out here in handleModal, line 62')
+      this.setState({
+        isHovered: true,
+        modalOpened: true
+      })
+    }
+
+    showModal () {
+      this.setState({
+        showingModal: !this.state.showingModal
+      })
+      console.log('logging from showModal, line 77')
+    }
+
+    hideModal () {
+      this.setState({
+        showingModal: false
+      })
+    }
+  
+    hoverHandler(e) {
+      let currentPhoto = e.currentTarget.src
+      this.setState({
+        isHovered: true,
+        mainphoto: currentPhoto
+      })
+      console.log(this.state.mainphoto, 'line 62')
+    }
+
+    hoverExit() {
+      this.setState({
+        isHovered: false
+      })
+      console.log(this.state.isHovered, 'line 71')
+    }
+
+    handleQuantityChange(e){
+      this.setState({
+        selectValue: e.target.value
+      });
+    }
+
+    hoverOptionHandler(e) {
+      console.log(e.target.value, 'logging from hoverOptionHandler')
+
+    }
+  
+    componentDidMount() {
+      this.getPhotos();
+      this.getMainProducts();
+      this.setState({
+        mainPhoto: this.state.photos[0]
+      })
+      if (this.state.isHovered) {
+        this.showModal()
+      }
+    }
+  
+    render() {
+      const { rating } = this.state;
+      return (
+        <div className="flex-container">
+          <div className="photos">
+            <Photos currentPhoto={this.state.mainphoto} hoverHandler={this.hoverHandler} hoverExit={this.hoverExit} isHovered={this.state.isHovered} photos={this.state.photos}/>
+          </div>
+          <div className='main-photo'>
+            <div className='main-photo-div'>
+              <MainPhoto showingModal={this.state.showingModal} hideModal={this.hideModal} showModal={this.showModal} modalOpened={this.state.modalOpened} mainPhoto={this.state.mainphoto} handleModal={this.handleModal}/>
+              <Modal mainPhoto={this.state.mainphoto} onClose={this.showModal} showingModal={this.state.showingModal}>
```

Comment:
could modal be a child component of mainphoto? Not sure if this state needs to all be passed down from the top or not, or if it could be contained in main photo.